aws4.js

```diff
@@ -27,6 +27,11 @@ function RequestSigner(request, credentials) {
   this.request = request
   this.credentials = credentials || this.defaultCredentials()
 
+  // ES's hostParts are the other way round
+  // E.g. search-cluster-name-aaaa00aaaa0aaa0aaaaaaa0aaa.us-east-1.es.amazonaws.com
```

Comment:
Same with here: http://docs.aws.amazon.com/general/latest/gr/rande.html#elasticsearch-service-regions

Are these configuration hosts, as opposed to service/data hosts? If so, should probably add a test to make sure they're supported as well (and that the hostnames aren't messed up)